sklearn/metrics/_plot/cap_curve.py

```diff
@@ -0,0 +1,344 @@
+            (self.chance_level_,) = self.ax_.plot(
+                x_vals, y_vals, **chance_level_line_kw
+            )
+        else:
+            self.chance_level_ = None
+
+        xlabel = "Total Cases Examined"
+        ylabel = "Cumulative True Positives"
+        self.ax_.set(xlabel=xlabel, ylabel=ylabel, aspect="equal")
+
+        if "label" in line_kwargs:
+            self.ax_.legend(loc="lower right")
+
+        return self
+
+    @classmethod
+    def from_predictions(
+        cls,
+        y_true,
+        y_pred,
+        *,
+        sample_weight=None,
+        pos_label=None,
+        normalize_scale=True,
+        plot_chance_level=True,
+        name=None,
+        ax=None,
+        **kwargs,
+    ):
+        """Plot the Cumulative Accuracy Profile.
+
+        This is also known as a (cumulative) gain curve.
+
+        Parameters
+        ----------
+        y_true : array-like of shape (n_samples,)
+            True labels.
+
+        y_pred : array-like of shape (n_samples,)
+            Target scores, can either be probability estimates of the positive
+            class, confidence values, or non-thresholded measure of decisions
+            (as returned by “decision_function” on some classifiers).
+
+        sample_weight : array-like of shape (n_samples,), default=None
+            Sample weights.
+
+        pos_label : int, float, bool or str, default=None
+            The label of the positive class. When `pos_label=None`, if `y_true`
+            is in {-1, 1} or {0, 1}, `pos_label` is set to 1, otherwise an
+            error will be raised.
+
+        normalize_scale : bool, default=True
+            Whether to normalize values between 0 and 1 for the plot.
+
+        name : str, default=None
+            Name of CAP curve for labeling. If `None`, name will be set to
+            `"Classifier"`.
+
+        ax : matplotlib axes, default=None
+            Axes object to plot on. If `None`, a new figure and axes is created.
+
+        **kwargs : dict
+            Additional keywords arguments passed to matplotlib `plot` function.
+
+        Returns
+        -------
+        display : :class:`~sklearn.metrics.CapCurveDisplay`
+            Object that stores computed values.
+        """
+        if pos_label is None:
+            pos_label = 1
+        if sample_weight is None:
+            sample_weight = np.ones_like(y_true, dtype=float)
+
+        # ensure y_true is boolean for positive class identification
+        y_bool = y_true == pos_label
+
+        # sort predictions and true values based on the predictions
+        sorted_indices = np.argsort(y_pred)[::-1]
+        y_true_sorted = y_bool[sorted_indices]
+        sample_weight_sorted = sample_weight[sorted_indices]
+
+        # compute cumulative sums for true positives and all cases
+        y_true_cumulative = np.cumsum(y_true_sorted * sample_weight_sorted)
+        cumulative_total = np.cumsum(sample_weight_sorted)
```

Comment:
For information, there was a concurrent PR to fix the lack of `cumsum` of the `sample_weight` to define the x-axis of the Lorenz curves in one of our regression examples. To check that this was the correct fix, we ran a quick check on synthetic data with integer valued sample weights to check that there is an exact equivalence between repeated data points and reweighting them by exposure:

https://github.com/scikit-learn/scikit-learn/pull/30198#issuecomment-2498615445

Maybe this PR could be expanded to test for this property also holds for `CapCurveDisplay.from_predictions` with non-constant, integer-valued `sample_weight`.